### ISSUE TITLE
Decouple platform's actor definition schema from catalog

### ISSUE THREAD

**Issue by pedroslopez** [airbytehq/airbyte#23578]
(rough ticket, wanted to start getting the idea out there)

## Tell us about the problem you're trying to solve

Currently, our CombinedConnectorCatalog consists of a list of StandardSourceDefinition and StandardDestinationDefinition. This was done at the start as an easy way to make sure the catalog contained things that the platform could load in.

As we move to multiple connector versions, we still want to provide this information via the catalog (for now) but on the platform side we will separate the StandardXYZDefinition classes from the ActorDefinitionVersion.

## Describe the solution you’d like

- Have a separate schema for the information provided by the catalog

**Comment by evantahler** [comment]
Grooming:
* This points to a shared library.  - this seems like https://github.com/airbytehq/airbyte-protocol
* These definitions should be versioned

**Comment by cpdeethree** [comment]
Going to discuss this ticket with charles and cole today and we will come back with a recommendation

**Comment by bnchrch** [comment]
@pedroslopez Im picking up this ticket!

Let me know if you want to sync over any thing before I get to deep.

My current planned approach is to

- [ ] Learn about @cpdeethree s plan for the `airbyte-protocol` repo
- [ ] Based on that make a recommendation as to where the jsonschema and subsequent types should live
- [ ] Implement that recommendation (pending feedback)
- [ ] Use the new catalog type definition in the `airbyte` repo
- [ ] Use the new catalog type definition in the `platform-internal` repo

**Comment by cpdeethree** [comment]
short answer is that we don’t want to reuse the protocol repo, but also that we don’t think airbyte-commons is the appropriate path forward. We have some ideas for multiple repositories but wanted to discuss with more people at the platform onsite. We also want to have a unified release/deployment strategy so having a bit more time to get that tooling in place makes sense

Our initial thought is more targeted repo's though. So this would probably go in some lib called airbyte-catalog rather than a generic airbyte-commons repository

so tl;dr can we hold off on what to do for a week?

**Comment by bnchrch** [comment]
Holding off a week will depend on @pedroslopez timeline I believe. But I dont imagine so.

I'm currently thinking this fits best as a published library so what repo its in shouldnt be a hard blocker for me

**Comment by bnchrch** [comment]
Following up here

My assumptions around the file impact "moving the definitions of `StandardXDefinition` inside  `CombinedConnectorCatalog` to their own catalog specific definitions" were off.

Looking at `StandardSourceDefinition` and `StandardDestinationDefinition` I found that we have a few references to other definitions that we will likely want to share with the catalog definitions. (AllowedHosts and NormalizationDestinationDefinitionConfig for example)

You can see which definitions we will likely share by looking [here](https://github.com/airbytehq/airbyte-types/tree/main/models/src/main/resources) in the poc repo I published.

@cpdeethree The number of files here leads me to believe it would be beneficial to have a separate "airbyte-types" repo that is responsible for generating the java and python files that represent the types commonly used across airbyte projects.

Im still thinking that `airbyte-protocol` is best left separate as the use case is slightly different and much more important.

Anyway I figured this [airbyte-types](https://github.com/airbytehq/airbyte-types) repo would be a good example of the "catalog to source definitions" for the platform onsite discussions next week.

@pedroslopez Given that the impact is a bigger that just the catalog, I'm going to pause work on this until we hear back from platform. With that in mind do you have an idea for how long until this issue becomes a blocker for you?

**Comment by pedroslopez** [comment]
Thanks for picking this up @bnchrch ! The planned approach looks good to me. 

In terms of timeframe, the piece that this blocks for the multiple versions project is removing the version-specific fields from the StandardSourceDefinition and StandardDestinationDefinition schemas. There are some pieces that have to be done before we get to this though, including some database changes and cleanup. Because there's still work to do, pushing this a week should be ok.

**Comment by bnchrch** [comment]
Great thanks for the context on timeline @pedroslopez .

I left a reminder for myself to checkin with @cconrad13 next wednesday. 

If their still isnt a clear idea to where we can host this type outside of `platform-internal` then we can just add the new ~Catalog~ Registry types to `platform-internal` and extract later.

**Comment by bnchrch** [comment]
Also @cgardens @evantahler this seems like a good place to have the debate of 
1. Stay with the Catalog name
2. Update to refer to these cloud and oss collections of Actor Definitions as a Registry.

**Comment by bnchrch** [comment]
@pedroslopez we are unblocked and have a green light for a airbyte-types repo from prodeng!

@evantahler Where can I get access to our pypi and maven central accounts to public packages?

**Comment by evantahler** [comment]
[Thought Doc](https://docs.google.com/document/d/1jl122NkgH9rs6mr58UgyOVqD9J4d_656BcCEMlVVfbQ/edit#heading=h.hxf91ev4axxq) on where metadata types should live

**Comment by evantahler** [comment]
Grooming:

* `airbytehq/airbyte-types` is dead.  Information moves into the `airbytehq/airbyte` repo + JSONschemas, publishing python and Java libs.
* short term plan is copy+paste between OSS and Platform repos
* future plan is that metadata will publish types to GCS and then platform will use it

**Comment by bnchrch** [comment]
Related library work: #24847

**Comment by pedroslopez** [comment]
closed by https://github.com/airbytehq/airbyte-platform-internal/pull/5849
